src/main/java/persistence/sql/ddl/type/H2DataTypeMapper.java

```diff
@@ -0,0 +1,32 @@
+package persistence.sql.ddl.type;
+
+import persistence.sql.ddl.exception.DataTypeNotFoundException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class H2DataTypeMapper implements DataTypeMapper {
+
+    private final Map<Class<?>, DataType> dataTypes = new HashMap<>();
```

Comment:
Map을 사용해서 성능도 함께 챙겨주셨네요! 👍 static 으로 선언해두면 더욱 좋을 것 같습니다 :)

```suggestion
    private static final Map<Class<?>, DataType> dataTypes = new HashMap<>();
```